### DISCUSSION TITLE
OKD 4.8 on online.net hosting baremetal

### DISCUSSION THREAD

**Discussion by Mateuy** [okd-project/okd#962] [Q&A]
Hello, I have checked discussions here but I didn't find any help for my issues. I was already looking for a solution or just tips on the web, but I think I have made a mistake in my configurations or made error in my process.

I have 5 servers, one for central required services (LD, DHCP, DNS, HTTP), one to bootstrap the cluster and 3 nodes to build a 3 CP cluster. My bootstrap server will be used for the first worker node after the cluster will boots right way.

All servers have 2 IP, a public IP provided by online.net and another one for my internal network on RPNv2 network to access and provision the cluster. 

The bootstrap process stopped at the message to "wait for 30m to bootstrap complete", and on each CP node, the hostname is not the one from internal DHCP but from online.net public IP DHCP service. I see that on k8s cluster not completely finished installation.

I think my compute nodes with eno1 (public IP) and eno2 (internal IP) don't boot correctly because eno1 take precedence each time there is reboot and bring fail into the bootstrap process for the 3 CP.

If anybody have details about that, I'lm happy to read you ;)

Thanks

Mateuy

**Comment by vrutkovs** [comment]
>The bootstrap process stopped at the message to "wait for 30m to bootstrap complete", and on each CP node, the hostname is not the one from internal DHCP but from online.net public IP DHCP service

See https://docs.okd.io/latest/installing/installing_bare_metal/installing-restricted-networks-bare-metal.html:
> If you use multiple NICs, specify a single interface in the ip option. For example, to use DHCP on a NIC that is named eno1, set ip=eno1:dhcp.

This interface would be configured using DHCP and get IP/hostname

**Reply by Mateuy** [reply]
Hi @vrutkovs,
Thank you for your reply, I set IP configurations into kernel boot options (by pressing TAB when FCOS live ISO is booting):
`ip=eno2.200:dhcp:9000`
**200** is the vlan ID to allow communication between all the nodes through internal network for provisionning.

I have **eno1** NIC which get DHCP configurations on public provider network and I think that OKD take **eno1** as primary NIC when bootstrap process (once controlplane reboot after ignition configurations) but I don't find any information or tips to make a setup like that. 

A little schema:

WAN <---> ENO1 <---> SERVERS <---> ENO2 <---> Internal Network for provisionning (LB, DHCP, DNS, HTTP) and bootstrap OKD

If you need more details, please let me know.
